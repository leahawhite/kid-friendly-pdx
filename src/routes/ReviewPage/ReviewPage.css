.ReviewPage {
  margin: 0 auto;
  max-width: 700px;
  display: flex;
  justify-content: center;
  align-items: center;
  flex-direction: column;
  min-height: calc(92vh - 79px);
}

.ReviewPage-header {
  text-align: center;
  font-size: 22px;
  font-weight: bold;
  margin: 40px 20px 0 20px;
}