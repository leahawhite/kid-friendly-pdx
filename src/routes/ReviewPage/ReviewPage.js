import React, { Component } from 'react'
import { Link } from 'react-router-dom'
import PlacesApiService from '../../services/places-api-service'
import Spinner from '../../components/Spinner/Spinner'
import ValidationError from '../../components/ValidationError/ValidationError'
import ReviewForm from '../../components/ReviewForm/ReviewForm'
import './ReviewPage.css'

export default class ReviewPage extends Component {
  static defaultProps = {
    place: {},
    insertReviewsAndImages: () => {}
  }

  state = {
    place: this.props.place || {},
    isLoading: false
  }

  componentDidMount() {
    const placeId = this.props.match.params.placeId
    if (!this.props.place.length) {
      this.setState({ isLoading: true })
      PlacesApiService.getById(placeId)
      .then(place => {
        this.setState({
          place,
          isLoading: false,
        })
      })
      .catch(error => {
        this.setState({ error: error })
      })
    }
  }
  render() {
    const { insertReviewsAndImages, error } = this.props
    const { place, isLoading } = this.state
    if (error) {
      return <ValidationError hasError={error} />
    } else if (isLoading) {
      return <Spinner />
    }
    return (
      <div className="ReviewPage">
        <h2 className="ReviewPage-header">Write a Review of{' '}
          <Link to={{
                    pathname: `/places/${place.id}`,
                    state: { place: place }
                    }}>
            {place.name}
          </Link>
        </h2>
        <ReviewForm place={place} insertReviewsAndImages={insertReviewsAndImages} error={error} />
      </div>
    )
  }
}