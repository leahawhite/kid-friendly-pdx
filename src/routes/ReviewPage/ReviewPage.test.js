import React from 'react';
import ReactDOM from 'react-dom';
import { MemoryRouter } from 'react-router-dom';
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import ReviewPage from './ReviewPage'

describe('ReviewPage component', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(
      <MemoryRouter>
        <ReviewPage />
      </MemoryRouter>, 
      div
    );
    ReactDOM.unmountComponentAtNode(div);
  });
  it('renders UI as expected', () => {
    const wrapper = shallow(
      <MemoryRouter>
        <ReviewPage />
      </MemoryRouter>)
    expect(toJson(wrapper)).toMatchSnapshot()
  });
})


