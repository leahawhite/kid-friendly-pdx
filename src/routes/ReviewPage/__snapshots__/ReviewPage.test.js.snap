// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ReviewPage component renders UI as expected 1`] = `
<Router
  history={
    Object {
      "action": "POP",
      "block": [Function],
      "canGo": [Function],
      "createHref": [Function],
      "entries": Array [
        Object {
          "hash": "",
          "key": "jv4rrg",
          "pathname": "/",
          "search": "",
          "state": undefined,
        },
      ],
      "go": [Function],
      "goBack": [Function],
      "goForward": [Function],
      "index": 0,
      "length": 1,
      "listen": [Function],
      "location": Object {
        "hash": "",
        "key": "jv4rrg",
        "pathname": "/",
        "search": "",
        "state": undefined,
      },
      "push": [Function],
      "replace": [Function],
    }
  }
>
  <ReviewPage
    location={
      Object {
        "state": Object {},
      }
    }
    place={Object {}}
  />
</Router>
`;
